doc/debugging.rst

```diff
@@ -14,33 +14,108 @@ This allows breakpoints and the execution flow to be viewed and controlled right
 Debugging with ipdb
 ===================
 
-Spyder offers the following debugging features integrated into the native GUI:
+You can fully control debugger execution from the :guilabel:`Debug` menu, :guilabel:`Debug toolbar` and via configurable keyboard shortcuts, along with the standard ``ipdb`` `console commands`_.
 
-* Multiple means of setting and clearing normal and conditional breakpoints for any line in a file opened in the :doc:`editor`.
+.. image:: images/debugging/debugging-commands.gif
+   :alt: A Spyder IPython console window, showing debugging from toolbar
 
-  * By selecting the respective option from the Debug menu.
-  * Through pressing a configurable keyboard shortcut (:kbd:`F12` for normal, or :kbd:`Shift-F12` for conditional breakpoints by default).
-  * By double-clicking to the left of the line number in an open file.
-  * With an ``ipdb.set_trace()`` statement in your code (after ``import`` ing ``pdb``).
-  * Interactively, using the ``b`` command in an ``ipdb`` session.
+
+Additionally, the  :doc:`editor` shows the line of code the debugger is currently stopped on with an arrow.
+
+.. _console commands: https://pythonconquerstheuniverse.wordpress.com/2009/09/10/debugging-in-python/
+
+.. image:: images/debugging/debugging-arrow.png
+   :alt: Spyder Editor showing the debugging panel
+
+
+Spyder's debugger offers syntax highlighting, code completion and command history, which work exactly like they do in the normal interactive interpreter.
+
+Use the up and down arrows to recall previous commands, and press :kbd:`Tab` to trigger autocomplete suggestions.
+
+.. image:: images/debugging/debugging-autocompletion.gif
+   :alt: A Spyder IPython console window, showing autocompletion when debugging
+
+
+Furthermore, IPython's `magic functions`_ are available in debugging mode.
+You can, for example, run ``%ls`` to list the contents of your current working directory or ``%timeit`` to check how fast a given snippet of code is.
+
+.. _magic functions: https://ipython.readthedocs.io/en/stable/interactive/magics.html
+
+.. image:: images/debugging/debugging-timeit.png
+   :alt: Ipython console in debug mode showing timeit magic
+
+Finally, you can enter and execute multiline statements in Spyder's debugger just like with the regular IPython prompt, to easily run complex code.
+
+.. image:: images/debugging/debugging-multiline.gif
```

Comment:
There's a bug in the code typed here, which the GIF makes clear: it prints `li` instead of `number`, printing the whole list three times instead of each number that you're iterating over, defeating the point of the loop. Instead, it should be `print(number)`.

Reply:
Agreed.

Reply:
This is now fixed, thanks.